tests/checks/mock/test_mesos_slave.py

```diff
@@ -0,0 +1,45 @@
+from tests.checks.common import AgentCheckTest, get_check_class, Fixtures
+
+from nose.plugins.attrib import attr
+from mock import patch
+from checks import AgentCheck
+import json
+import time
+
+
+def _mocked_get_state(*args, **kwargs):
+    state = json.loads(Fixtures.read_file('state.json'))
+    return state
+def _mocked_get_stats(*args, **kwargs):
+    stats = json.loads(Fixtures.read_file('stats.json'))
+    return stats
+
+@attr(requires='mesos_slave')
+class TestMesosSlave(AgentCheckTest):
+    CHECK_NAME = 'mesos_slave'
+
+    def test_checks(self):
+        config = {
+            'init_config': {},
+            'instances': [
+                {
+                    'url': 'http://localhost:5050',
+                    'tasks': ['hello']
+                }
+            ]
+        }
+
+        klass = get_check_class('mesos_slave')
+        with patch.object(klass, '_get_state', _mocked_get_state):
+            with patch.object(klass, '_get_stats', _mocked_get_stats):
+                check = klass('mesos_slave', {}, {})
+                self.run_check(config)
+                time.sleep(1)
+                self.run_check(config)
```

Comment:
Tiny nitpick, you can use the run_check_twice helper instead
https://github.com/DataDog/dd-agent/blob/master/tests/checks/common.py#L160